tests/storage/test_devices.py

```diff
@@ -71,6 +71,28 @@ def test_get_devices_by_user(self):
             res["device2"],
         )
 
+    @defer.inlineCallbacks
+    def test_get_devices_by_remote(self):
+        device_ids = ["device_id1", "device_id2"]
+
+        # Add a device update to the stream
+        yield self.store.add_device_change_to_streams(
+            "user_id", device_ids, ["somehost"],
+        )
+
+        # Get all device updates ever meant for this remote
+        res = yield self.store.get_devices_by_remote("somehost", -1, limit=100)
+
+        device_updates = res[1]
+
+        for update in device_updates:
```

Comment:
can we assert that there are exactly two updates?

Reply:
The code does unless you want to replace it with just an assertion of two updates.

Reply:
the code does not appear to assert that currently?

Reply:
Ah I see, it asserts there's at least two, but not that there's only two. Will update.